kernel/kernel-api/src/main/java/io/delta/kernel/config/ConfigurationProvider.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.config;
+
+import io.delta.kernel.annotation.Evolving;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+/**
+ * A generic interface for retrieving configuration values.
+ *
+ * @since 3.3.0
+ */
+@Evolving
+public interface ConfigurationProvider {
```

Comment:
@scottsand-db  This class doesn't provide any method to iterate all configs. Does that work for UC Commit Coordinator? Don't we need to iterate over all configs in that? Will let @sumeet-db confirm.

Reply:
@prakharjain09 -- Yup, great callout. We can update APIs when we get there. IMO there's no rush or requirement to add it now.

Reply:
Will need a `getAll` / `getAllByPrefix` API for UC Commit Coordinator Builder when we get there.